dataland-frontend/tests/component/components/resources/frameworkDataSearch/lksg/LksgPanel.cy.ts

```diff
@@ -167,25 +167,42 @@ describe("Component test for LksgPanel", () => {
     const secondOtherObject = { dataId: "6", reportingPeriod: "Q3-2020" };
     const boolList = [false, true]; //Apparently Typescript doesn't like type conversions, so input is direct.
     for (let i = 0; i < 2; i++) {
-      expect(sortReportingPeriodsToDisplayAsColumns([secondYearObject, firstYearObject], boolList[i])).to.deep.equal([
-        firstYearObject,
-        secondYearObject,
-      ]);
-
-      expect(sortReportingPeriodsToDisplayAsColumns([secondOtherObject, firstOtherObject], boolList[i])).to.deep.equal([
-        firstOtherObject,
-        secondOtherObject,
-      ]);
-    }
+      expect(
+        swapAndSortReportingPeriodsToDisplayAsColumns([secondYearObject, firstYearObject], boolList[i])
+      ).to.deep.equal([firstYearObject, secondYearObject]);
 
+      expect(
+        swapAndSortReportingPeriodsToDisplayAsColumns([secondOtherObject, firstOtherObject], boolList[i])
+      ).to.deep.equal([firstOtherObject, secondOtherObject]);
+    }
     expect(sortReportingPeriodsToDisplayAsColumns([firstYearObject, secondYearObject, firstYearObject])).to.deep.equal([
       firstYearObject,
       firstYearObject,
       secondYearObject,
     ]);
-
     expect(
       sortReportingPeriodsToDisplayAsColumns([firstYearObject, secondOtherObject, firstOtherObject])
     ).to.deep.equal([firstYearObject, firstOtherObject, secondOtherObject]);
   });
 });
+
+/**
+ * Calls the testfunction for sorting and swaps the list entries if necessary.
+ * @param  listOfDataDateToDisplayAsColumns list of objects to sort
+ * @param boolSwap toogles the swap of both list elements in listOfDataDateToDisplayAsColumns (in case there are two.
```

Comment:
Why is the name `boolSwap`? Why does it incorporate `bool`? I find `swap` or `shouldSwap` would be fine.

Reply:
done